src/AiClient.php

```diff
@@ -0,0 +1,421 @@
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the model doesn't support any known generation type.
+     */
+    public static function generateResult($prompt, ModelInterface $model): GenerativeAiResult
+    {
+        // Simple type checking instead of over-engineered resolver
+        if ($model instanceof TextGenerationModelInterface) {
+            return self::generateTextResult($prompt, $model);
+        }
+
+        if ($model instanceof ImageGenerationModelInterface) {
+            return self::generateImageResult($prompt, $model);
+        }
+
+        if ($model instanceof TextToSpeechConversionModelInterface) {
+            return self::convertTextToSpeechResult($prompt, $model);
+        }
+
+        if ($model instanceof SpeechGenerationModelInterface) {
+            return self::generateSpeechResult($prompt, $model);
+        }
+
+        throw new \InvalidArgumentException(
+            'Model must implement at least one supported generation interface ' .
+            '(TextGeneration, ImageGeneration, TextToSpeechConversion, SpeechGeneration)'
+        );
+    }
+
+    /**
+     * Creates a new message builder for fluent API usage.
+     *
+     * This method will be implemented once MessageBuilder is available.
+     * MessageBuilder will provide a fluent interface for constructing complex
+     * messages with multiple parts, attachments, and metadata.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|null $text Optional initial message text.
+     * @return object MessageBuilder instance (type will be updated when MessageBuilder is available).
+     *
+     * @throws \RuntimeException When MessageBuilder is not yet available.
+     */
+    public static function message(?string $text = null)
+    {
+        throw new \RuntimeException(
+            'MessageBuilder is not yet available. This method depends on builder infrastructure. ' .
+            'Use direct generation methods (generateTextResult, generateImageResult, etc.) for now.'
+        );
+    }
+
+    /**
+     * Template method for executing generation operations.
+     *
+     * NOTE: This method currently uses PromptNormalizer directly, but will be refactored
+     * to delegate to PromptBuilder once PR #49 is merged, following the architectural
+     * pattern where traditional API methods wrap the fluent builder API.
+     *
+     * @since n.e.x.t
+     *
+     * @param Prompt $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @param string $type The generation type.
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the prompt format is invalid.
+     * @throws \RuntimeException If no suitable model is found.
+     */
+    private static function executeGeneration($prompt, ?ModelInterface $model, string $type): GenerativeAiResult
```

Comment:
Related to my above comment, I don't think this is the right abstraction. No need to pass a `$type` - the model config's `outputModalities` should determine what to generate.

Passing a type duplicates what those are supposed to be for, but more importantly they prevent the implementation of multimodal output (it wouldn't be always just one `$type`).